sale_product_set/models/product_set_line.py

```diff
@@ -15,8 +16,8 @@ def prepare_sale_order_line_values(self, order, quantity, max_sequence=0):
             "order_id": order.id,
             "product_id": self.product_id.id,
             "product_uom_qty": self.quantity * quantity,
-            "product_uom": self.product_id.uom_id.id,
+            "product_uom": self.product_id and self.product_id and self.product_id.uom_id and self.product_id.uom_id.id or False,
```

Comment:
You can simplify by:

```suggestion
            "product_uom": self.product_id.uom_id and self.product_id.uom_id.id or False,
```

Reply:
Even if self.product_id.uom_id.id will be False if any relation is not defined. So, I'm wondering in which case you'll have an error